runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -193,17 +193,30 @@ jvmtiGetThreadListStackTraces(jvmtiEnv* env,
 	jint max_frame_count,
 	jvmtiStackInfo** stack_info_ptr)
 {
-	J9JavaVM * vm = JAVAVM_FROM_ENV(env);
-	jvmtiError rc;
-	J9VMThread * currentThread;
-	PORT_ACCESS_FROM_JAVAVM(vm);
+	J9JavaVM *vm = JAVAVM_FROM_ENV(env);
+	jvmtiError rc = JVMTI_ERROR_NONE;
+	J9VMThread *currentThread = NULL;
 	jvmtiStackInfo *rv_stack_info = NULL;
+	PORT_ACCESS_FROM_JAVAVM(vm);
 
 	Trc_JVMTI_jvmtiGetThreadListStackTraces_Entry(env);
 
 	rc = getCurrentVMThread(vm, &currentThread);
 	if (rc == JVMTI_ERROR_NONE) {
-		jvmtiStackInfo * stackInfo;
+		jvmtiStackInfo *stackInfo;
+		const jint originalThreadCount = thread_count;
+		const jthread *originalThreadList = thread_list;
+
+		while (0 != thread_count) {
+			jthread thread = *thread_list;
+			J9VMThread *targetThread = NULL;
+
+			if ((NULL != thread) && IS_VIRTUAL_THREAD(currentThread, J9_JNI_UNWRAP_REFERENCE(thread))) {
+				getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
+			}
+			++thread_list;
+			--thread_count;
+		}
 
 		vm->internalVMFunctions->internalEnterVMFromJNI(currentThread);
```

Comment:
this have to be before the while loop, also move all the `ENSURE_...` before the while loop as well